File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
       unordered_map<int,vector<int>>b;
       for (int i=0;i<nums1.size();i++){
           if (find(nums2.begin(),nums2.end(),nums1[i])!=nums2.end()){
               b[nums1[i]] = {(int)count(nums1.begin(),nums1.end(),nums1[i]),(int)count(nums2.begin(),nums2.end(),nums1[i])};
           }
       }
       vector<int>x;
       for (auto [key,val] : b){
           for (int i=0;i<min(val[0],val[1]);i++){
               x.push_back(key);
           }
       }
       return x;

        
    }
};

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.py
class Solution(object):
    def intersect(self, nums1, nums2):
        b={}
        for i in range(len(nums1)):
            if nums1[i] in nums2:
                b[nums1[i]]=[nums1.count(nums1[i]),nums2.count(nums1[i])]
        c=[]
        for key,val in b.items():
            for i in range(min(val)):
                c.append(key) 
        return c
    